bgpd/bgp_io.c

```diff
@@ -464,10 +495,18 @@ static uint16_t bgp_read(struct peer *peer, int *code_p)
 {
 	size_t readsize; // how many bytes we want to read
 	ssize_t nbytes;  // how many bytes we actually read
+	size_t ibuf_work_space; // how much space we can read into the work buf
 	uint16_t status = 0;
 
-	readsize =
-		MIN(ringbuf_space(peer->ibuf_work), sizeof(peer->ibuf_scratch));
+	ibuf_work_space = ringbuf_space(peer->ibuf_work);
+
+	if (ibuf_work_space == 0) {
+		SET_FLAG(status, BGP_IO_WORK_FULL_ERR);
```

Comment:
Should we consider adding a warning/log for an operator to see that they have hit this condition?  Might be useful for an operator to be aware they're hitting a memory limit so they can investigate a condition causing bgpd to be slow about reading from the inQ.

Reply:
i had one, but it's extremely chatty

Reply:
if we hit this conditional at load, it will print everytime we read from the socket basically while we wait for room

Reply:
Yeah, if we want to log the condition then we'd have be smart about making sure it wouldn't be overwhelming in those conditions.

Reply:
fixed, used static bool so its only printed once and cleared when inqueue is drained